server/mediaserver.go

```diff
@@ -529,10 +540,13 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 	body, err := ioutil.ReadAll(r.Body)
 
 	if err != nil {
-		http.Error(w, fmt.Sprintf(`Error reading http request body: %s`, err.Error()), http.StatusInternalServerError)
+		httpErr := fmt.Sprintf(`Error reading http request body: %s`, err.Error())
+		glog.Error(httpErr)
+		http.Error(w, httpErr, http.StatusInternalServerError)
 		return
 	}
 	r.Body.Close()
+	r.URL = &url.URL{Scheme: "http", Host: r.Host, Path: r.URL.Path}
```

Comment:
Can unit tests be added to check this bit works as expected?

Reply:
I'm not sure how to test that - whether I include this or not: `r.URL = &url.URL{Scheme: "http", Host: r.Host, Path: r.URL.Path}` .... both tests will either pass or not. Any suggestions?

Reply:
I've tried editing the request URL here, `req := httptest.NewRequest("GET", "/live/seg.ts", reader)`, trying different permutations, and with or without the line in question, the result is the same (either passing or not passing)

Reply:
These give me the same values: 
```
glog.Error("R URL Scheme: ", r.URL.Scheme)
	glog.Error("R URL Host: ", r.URL.Host)
	glog.Error("R URL Host222: ", r.Host)
	glog.Error("R URL Port: ", r.URL.Port())
	glog.Error("R URL Path: ", r.URL.Path)
	glog.Error("R URL Query: ", r.URL.Query())
	glog.Error("R requestURI", r.RequestURI)
	glog.Error("R header HOST: ", r.Header.Get("Host"))
	glog.Error("R header forwarded proto: ", r.Header.Get("X-Forwarded-Proto"))
```

Reply:
> both tests will either pass or not

Which tests?

The original concern around this line was that the URL in the webhook request did not have the full scheme or host. So I'd build the test along those lines - stub out the webhook callback, invoke the push, check the URL in the callback request matches.

If you look at https://golang.org/pkg/net/http/httptest/#NewRequest then the documentation states that if the host is not given, the `example.com` is used. So for your example of `req := httptest.NewRequest("GET", "/live/seg.ts", reader)` , the URL in the webhook should look something like `http://example.com/live/seg.ts`